tide/core/LayoutEngine.cpp

```diff
@@ -152,7 +153,10 @@ QRectF LayoutEngine::_getNominalCoord( const ContentWindow& window ) const
 
     QSizeF size = window.getCoordinates().size();
     size.scale( maxSize, Qt::KeepAspectRatio );
-    window.getController()->constrainSize( size );
+    ContentWindowController( const_cast<ContentWindow&>( window ),
```

Comment:
the window is not supposed to be modified from here. Providing another constructor seems better.